evm-ds/src/evm_server_run.rs

```diff
@@ -27,7 +27,7 @@ use crate::tracing_logging::{CallContext, LoggingEventListener};
 fn build_exit_result<B: Backend>(
     executor: CpsExecutor<B>,
     runtime: &Runtime,
-    trace: &LoggingEventListener,
+    trace: Arc<Mutex<LoggingEventListener>>,
```

Comment:
This was changed so that before you start, you create this listener, and all evm executions point to and build on the same one, essentially.

Reply:
The listener was changed from how it was because previously it was ser/deser when communicating with the cpp code, but actually we don't need that overhead.

A slightly awkward thing is that it needs to be present in both the args, so the evm sees it, and in the result, so it gets passed back to consensus.